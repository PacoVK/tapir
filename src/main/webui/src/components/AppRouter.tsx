import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import ModuleOverview from "../views/ModuleOverview";
import ModuleDetails from "../views/ModuleDetails";
import ProviderOverview from "../views/ProviderOverview";
import ProviderDetails from "../views/ProviderDetails";
import ManagementView from "../views/ManagementView";
import { useUserContext } from "./context/UserContext";
import NotFoundPage from "../views/404";

const AppRouter = () => {
  const { isAdmin } = useUserContext();
  return (
      <Routes>
        <Route path="/" element={<ModuleOverview />} />
        <Route path="/providers" element={<ProviderOverview />} />
        {isAdmin ? (
          <Route path="/management" element={<ManagementView />} />
        ) : null}
        <Route
          path="/module/:namespace/:name/:provider"
          loader={({ params }) => {
            return fetch(
              `http://localhost:8080/terraform/modules/v1/${params.namespace}/${params.name}/${params.provider}`,
            );
          }}
          element={<ModuleDetails />}
        />
        <Route
          path="/providers/:namespace/:type"
          loader={({ params }) => {
            return fetch(
              `http://localhost:8080/terraform/providers/v1/${params.namespace}/${params.type}`,
            );
          }}
          element={<ProviderDetails />}
        />
        <Route path={"*"} element={<NotFoundPage />} />
      </Routes>
  );
};

export default AppRouter;
