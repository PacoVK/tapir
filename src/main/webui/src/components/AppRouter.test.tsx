import React from "react";
import { render } from "@testing-library/react";
import AppRouter from "./AppRouter";
import { BrowserRouter } from "react-router-dom";

describe("<AppRouter /> spec", () => {
  it("renders the AppRouter", () => {
    const view = render(
      <BrowserRouter
        future={{
          v7_startTransition: true,
          v7_relativeSplatPath: true,
        }}
      >
        <AppRouter />
      </BrowserRouter>,
    );
    expect(view).toMatchSnapshot();
  });
});
