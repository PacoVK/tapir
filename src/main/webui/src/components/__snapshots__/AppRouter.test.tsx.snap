// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> spec renders the AppRouter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="module-search"
          id="module-search-label"
        >
          Search module
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch css-7pxcz1-MuiInputBase-input-MuiOutlinedInput-input"
            id="module-search"
            type="search"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Search module
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiStack-root MuiList-root MuiList-padding css-1j5kjsk-MuiStack-root-MuiList-root"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-1gcdw72-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="module-search"
        id="module-search-label"
      >
        Search module
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch css-7pxcz1-MuiInputBase-input-MuiOutlinedInput-input"
          id="module-search"
          type="search"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-81qg8w"
          >
            <span>
              Search module
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiStack-root MuiList-root MuiList-padding css-1j5kjsk-MuiStack-root-MuiList-root"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-1gcdw72-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
